File: algorithms/strings/run-length-encoding/swift/RunLengthEncoding.swift
func runLengthEncoding(_ arr: [Int]) -> [Int] {
    if arr.isEmpty { return [] }
    var result: [Int] = []
    var count = 1
    for i in 1..<arr.count {
        if arr[i] == arr[i-1] { count += 1 }
        else { result.append(contentsOf: [arr[i-1], count]); count = 1 }
    }
    result.append(contentsOf: [arr.last!, count])
    return result
}

File: algorithms/strings/run-length-encoding/kotlin/RunLengthEncoding.kt
fun runLengthEncoding(arr: IntArray): IntArray {
    if (arr.isEmpty()) return intArrayOf()
    val result = mutableListOf<Int>()
    var count = 1
    for (i in 1 until arr.size) {
        if (arr[i] == arr[i-1]) count++
        else { result.add(arr[i-1]); result.add(count); count = 1 }
    }
    result.add(arr.last()); result.add(count)
    return result.toIntArray()
}
